Modules/_sqlite/connection.c

```diff
@@ -188,17 +154,22 @@ _sqlite3.Connection.__init__ as pysqlite_connection_init
 [clinic start generated code]*/
 
 static int
-pysqlite_connection_init_impl(pysqlite_Connection *self,
-                              const char *database, double timeout,
-                              int detect_types, const char *isolation_level,
+pysqlite_connection_init_impl(pysqlite_Connection *self, PyObject *database,
+                              double timeout, int detect_types,
+                              const char *isolation_level,
                               int check_same_thread, PyObject *factory,
                               int cache_size, int uri)
-/*[clinic end generated code: output=7d640ae1d83abfd4 input=342173993434ba1e]*/
+/*[clinic end generated code: output=839eb2fee4293bda input=b8ce63dc6f70a383]*/
 {
     if (PySys_Audit("sqlite3.connect", "s", database) < 0) {
         return -1;
     }
 
+    PyObject *bytes;
+    if (!PyUnicode_FSConverter(database, &bytes)) {
+        return -1;
+    }
```

Comment:
Regarding https://github.com/python/cpython/pull/92926#issuecomment-1131475986, I think a short comment above line 168 on why this need to come after `PySys_Audit` would be nice.

Reply:
I do not have a good comment which would survive the time. In any case, `test_open_with_undecodable_path` fails without these changes, so it is unlikely that somebody will incidentally reintroduce the same bug.

Reply:
Good point.